File: Leetcode 2901-3000/2996/2996.java
// Leetcode 2996: Smallest Missing Integer Greater Than Sequential Prefix Sum
// https://leetcode.com/problems/smallest-missing-integer-greater-than-sequential-prefix-sum/
// Solved on 11th of August, 2026
class Solution {
    /**
     * Calculates the smallest missing integer greater than the sequential prefix sum of the array.
     * @param nums The input array of integers.
     * @return The smallest missing integer greater than the sequential prefix sum.
     */
    public int missingInteger(int[] nums) {
        int sum = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == nums[i - 1] + 1) {
                sum += nums[i];
            } else {
                break;
            }
        }
        boolean[] present = new boolean[51];
        for (int num : nums) {
            present[num] = true;
        }
        while (sum <= 50 && present[sum]) {
            sum++;
        }
        return sum;
    }
}

File: Leetcode 2901-3000/2996/2996.py
# Leetcode 2996: Smallest Missing Integer Greater Than Sequential Prefix Sum
# https://leetcode.com/problems/smallest-missing-integer-greater-than-sequential-prefix-sum/
# Solved on 11th of August, 2026
class Solution:
    def missingInteger(self, nums: list[int]) -> int:
        """
        Calculates the smallest missing integer greater than the sequential prefix sum of the array.
        
        @param nums: The input array of integers.
        @return: The smallest missing integer greater than the sequential prefix sum.
        """
        prefixSum = nums[0]
        arrayLength = len(nums)

        for index in range(1, arrayLength):
            if nums[index] == nums[index - 1] + 1:
                prefixSum += nums[index]
            else:
                break

        numSet = set(nums)
        targetValue = prefixSum

        while targetValue in numSet:
            targetValue += 1

        return targetValue